backstop_data/engine_scripts/puppet/page-objects/crm-page.js

```diff
@@ -42,11 +42,14 @@ module.exports = class CrmPage {
    * Closes all active notifications.
    */
   async closeErrorNotifications () {
-    const exists = !!(await this.engine.$('.ui-notify-message'));
-
-    if (exists) {
-      await this.engine.click('a.ui-notify-cross.ui-notify-close');
-      await this.engine.waitFor('.ui-notify-message', { hidden: true });
+    while(1) {
+      const exists = !!(await this.engine.$('.ui-notify-message'));
```

Comment:
if the problem was that there are multiple notifications, then puppeteer provides the [`.$$`](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pageselector-1) method, which is the equivalent of `document.querySelectorAll()`.

You could use it to know how many notifications are there to be closed, and then set up a loop to close as many notifications as `.$$` found. I'd suggest to process the list in sequence (see [here](https://blog.lavrton.com/javascript-loops-how-to-handle-async-await-6252dd3c795), to make sure we have a reliable result

```js
const notifications = this.engine.$$('.ui-notify-message');

if (notifications.length) {
  for (const notification of notifications) {
    await this.engine.click('a.ui-notify-cross.ui-notify-close');
    await this.engine.waitFor('.ui-notify-message', { hidden: true });
  }
}
```
Notice that we don't really use the `notification` object in this case, as we're using the `notifications` array only to know *how many* notifications there are to close, not *which one* we are closing at any given time